File: TheWorld/src/TheWorld/Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using TheWorld.Model;
using TheWorld.ViewModels;

namespace TheWorld.Controllers.Api
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        private ILogger<TripsController> _logger;
        private IWorldRepository _repository;

        public TripsController(IWorldRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }
        [HttpGet("")]
        public IActionResult Get()
        {
            try
            {
                var trips = _repository.GetUserTripsWithStops(User.Identity.Name);
                var results = Mapper.Map<IEnumerable<TripViewModel>>(trips);
                return Ok();
            }
            catch (Exception ex)
            {
                _logger.LogError($"Faield to get all trips { ex }");
                return BadRequest("Error Occurred");
            }

        }
        [HttpPost("")]
        public async Task<IActionResult> Post([FromBody]TripViewModel theTrip)
        {
            if (ModelState.IsValid)
            {
                // save to db
                var newTrip = Mapper.Map<Trip>(theTrip);
                newTrip.UserName = User.Identity.Name;
                _repository.AddTrip(newTrip);
                if (await _repository.SaveChangesAsync())
                {
                    return Created($"api/trips/{theTrip.Name}", Mapper.Map<TripViewModel>(newTrip));
                }
                else
                {
                    return BadRequest("Failed to save changes");
                }
            }

            return BadRequest(ModelState);
        }
    }
}

File: TheWorld/wwwroot/js/tripsController.js
(function() {
    "use strict";
    angular.module("app-trips")
    .controller("tripsController", tripsController);

    function tripsController($http) {
        var vm = this;
        vm.trips = [{
            name: "US trip",
            created: new Date()
        },
        {
            name: "World trip",
            created: new Date()
        }];

        vm.newTrip = {};
        vm.isBusy = true;
        vm.errorMessage = "";

        $http.get("/api/trips")
        .then(function (response) {
            angular.copy(response.data, vm.trips);
            vm.isBusy = false;
        }, function() {
            vm.errorMessage = "Failed to load data" + error;
        })
        .finally(function() {
            vm.isBusy = false;
        });

        vm.addTrip = function () {
            vm.isBusy = true;
            vm.errorMessage = "";
            $http.post("api/trips", vm.newTrip)
            .then(function(response) {
                vm.trips.push(response.data);
                vm.newTrip = {};
            }, function() {
                vm.errorMessage = "failed to save new trip";
            })
            .finally(function() {
                vm.isBusy = false;
            })
            ;
        };
    }


})();